htlcswitch/switch.go

```diff
@@ -566,37 +576,60 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 	// payment circuit by forwarding the settle msg to the channel from
 	// which htlc add packet was initially received.
 	case *lnwire.UpdateFufillHTLC, *lnwire.UpdateFailHTLC:
-		// Exit if we can't find and remove the active circuit to
-		// continue propagating the fail over.
-		circuit, err := s.circuits.remove(packet.payHash)
-		if err != nil {
-			err := errors.Errorf("unable to remove "+
-				"circuit for payment hash: %v", packet.payHash)
-			log.Error(err)
-			return err
+		if !packet.isRouted {
```

Comment:
Ahh was commenting on an out of date diff.

Reply:
Would be worth adding a comment here stating that things are only routed once they've reached the _outgoing_ channel. So in the case that we initiate a local payment, then there's nothing in the circuit map to delete.

Reply:
That's not what `isRouted` refers to. `isRouted` means the incomingChanID and incomingHTLCID are already set and do not need to be looked up in the circuit map. Local payments still get added and removed from the circuit map.